test/boost/cql_query_test.cc

```diff
@@ -4313,3 +4313,55 @@ SEASTAR_TEST_CASE(test_rf_expand) {
         });
     });
 }
+
+SEASTAR_TEST_CASE(test_bigint_sum) {
+    return do_with_cql_env_thread([] (cql_test_env& e) {
+        cquery_nofail(e, "create table cf (pk text, val bigint, primary key(pk));");
+        cquery_nofail(e, "insert into cf (pk, val) values ('x', 2147483647);");
+        cquery_nofail(e, "insert into cf (pk, val) values ('y', 2147483647);");
+        assert_that(e.execute_cql("select sum(val) from cf;").get0())
+            .is_rows()
+            .with_size(1)
+            .with_row({long_type->decompose(int64_t(4294967294))});
+
+        cquery_nofail(e, "insert into cf (pk, val) values ('z', -4294967295);");
+        assert_that(e.execute_cql("select sum(val) from cf;").get0())
+            .is_rows()
+            .with_size(1)
+            .with_row({long_type->decompose(int64_t(-1))});
+    });
+}
+
+SEASTAR_TEST_CASE(test_int_sum_overflow) {
```

Comment:
This tests already pass is non debug builds, which suggests that std::make_unsigned is the way to go.

Reply:
I agree from a programming perspective, but not sure if it's the right thing to do from the usability perspective.